src/content/blog/shorebird-code-push-v2.md

```diff
@@ -0,0 +1,161 @@
+---
+title: What's New in v2.0 of the Shorebird Code Push Package
+author: felangel
+description: Announcing the pre-release of pkg:shorebird_code_push v2.0
+date: 2024-11-04
+cover: /blog/shorebird-code-push-v2-cover.png
+coverAlt: Improved Patch Delivery Cover Image
+---
+
+We're excited to announce a pre-release of version 2.0 of the
+[`shorebird_code_push`
+package](https://pub.dev/packages/shorebird_code_push/versions/2.0.0-dev.2).
+
+## What's New
+
+We've taken some time to re-imagine what the Dart API should look like when
+interfacing with the Shorebird updater and we've re-written the Dart API from the
+ground up.
```

Comment:
I would remind people that use of the API is completely optional.  Most of our customers don't use it and Shorebird works great for them.

Reply:
Maybe we need a bit that explains when you would use the API?

e.g. a custom update flow to do...

1. Only send updates to some users
2. Force users to update.
3. Download and update via background message (to get updates sooner than restarting).
4. ... more?

Reply:
And then explain what UX you're explaining in this blog post.  e.g. In example X (link) we show how you might do Y (where hopefully Y is one of the above, and some usecase we think is valid).

e.g. if we're jsut showing them how to show a dialog when updating, that feels bad for the user?

Maybe 5. is "Record additiona information about update success in the wild to your analytics provider"